### DISCUSSION TITLE
请问如何安装自定义的Python组件？

### DISCUSSION THREAD

**Discussion by thomas-lint** [flagify-com/OctoMation#27] [Q&A] [answered]
我的Python程序有一些第三方的包，需要通过pip安装，这种该怎么操作？

**Comment by ghost** [comment, marked as answer]
可以进入执行引擎所在的容器安装 
docker exec -it shakespeare-java-exe-python /bin/bash
pip install xxx

**Comment by wuzhi-dev** [comment]
Tips：根据目前的机制，未来docker镜像升级后，不影响已经安装的包文件。

**Comment by wuzhi-dev** [comment]
参考wiki:https://github.com/flagify-com/OctoMation/wiki/OctoMation%E4%B8%AD%E4%BD%BF%E7%94%A8%E7%AC%AC%E4%B8%89%E6%96%B9Python%E6%A8%A1%E5%9D%97
